source/backend/api/Services/PropertyService.cs

```diff
@@ -209,9 +209,70 @@ public PropertyManagementModel GetPropertyManagement(long propertyId)
             propertyManagement.RelatedLeases = leaseCount;
             propertyManagement.LeaseExpiryDate = leaseExpiryDate.HasValue ? DateOnly.FromDateTime(leaseExpiryDate.Value) : null;
 
+            PropertyHasActiveLease(propertyLeases, out bool hasActiveLease, out bool hasActiveExpiryDate);
+            propertyManagement.HasActiveLease = hasActiveLease;
+            propertyManagement.ActiveLeaseHasExpiryDate = hasActiveExpiryDate;
+
             return propertyManagement;
         }
 
+        private static void PropertyHasActiveLease(IEnumerable<PimsPropertyLease> propertyLeases, out bool hasActiveLease, out bool hasActiveExpiryDate)
+        {
+            hasActiveLease = false;
+            hasActiveExpiryDate = false;
+
+            List<PimsLease> activeAgreementsList = propertyLeases.Select(x => x.Lease).ToList();
+            var activeLease = activeAgreementsList.FirstOrDefault(x => x.LeaseStatusTypeCode == LeaseStatusTypes.ACTIVE.ToString());
```

Comment:
My understanding of the requirements, is that if there is **any** lease that is active for a property, that should be displayed on the management screen.

Given that a single property may have multiple leases (ie, a large lot with many buildings). it seems entirely possible that a single property may have multiple active leases, where some of those active leases may have expired or been terminated.

as a result, I think using FirstOrDefault here doesn't work.

Reply:
Updated to check for other leases with a loop.